src/packet_builder.rs

```diff
@@ -1200,6 +1200,16 @@ impl PacketBuilderStep<IpHeader> {
             _marker: marker::PhantomData::<TcpHeader>{}
         }
     }
+
+    ///Write all the headers and the payload.
+    pub fn write<T: io::Write + Sized>(self, writer: &mut T, payload: &[u8]) -> Result<(),WriteError> {
```

Comment:
I would add the last ip_number as an argument here so it can be set in the last extension header or ip header.